src/codegen/barrier.cpp

```diff
@@ -0,0 +1,39 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// barrier.cpp
+//
+// Identification: src/codegen/barrier.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "common/logger.h"
+#include "codegen/barrier.h"
+#include "codegen/utils/oa_hash_table.h"
+
+namespace peloton {
+namespace codegen {
+
+void Barrier::InitInstance(Barrier *ins, uint64_t n_workers)
+{
+    ins->SetBarrier(new boost::barrier(n_workers));
```

Comment:
Did you tried valgrind? Will this cause a memory leak? Will it be better if you use a smart ptr?